src/node_zlib.cc

```diff
@@ -995,15 +1012,14 @@ CompressionError ZlibContext::Init(
       UNREACHABLE();
   }
 
-  dictionary_ = std::move(dictionary);
-
   if (err_ != Z_OK) {
     dictionary_.clear();
     mode_ = NONE;
-    return ErrorForMessage("zlib error");
+    return true;
```

Comment:
What happens if there is an error during initialization? (I assume this would mostly be relevant for out-of-memory situations?)

Reply:
Do you mean `ZlibContext::Init()` or `ZlibContext::InitZlib()`? If that's the latter, then the error should be emitted after `ZlibContext::DoThreadPoolWork()` execution finishes (which should be done in `AfterThreadPoolWork()` in case of the async execution).

Reply:
Right, thanks for pointing that out :)